modules/sgct/src/datastructures/sgctcamera.cpp

```diff
@@ -0,0 +1,194 @@
+
+#include <sgct/sgct.h>
+
+namespace inviwo {
+
+SGCTCamera::SGCTCamera(vec3 lookFrom, vec3 lookTo, vec3 lookUp, float nearPlane, float farPlane,
+                       float fieldOfView, float aspectRatio)
+    : Camera(lookFrom, lookTo, lookUp, nearPlane, farPlane, aspectRatio), fovy_(fieldOfView) {}
+
+SGCTCamera::SGCTCamera(const SGCTCamera& other) = default;
+
+SGCTCamera& SGCTCamera::operator=(const SGCTCamera& other) = default;
+
+SGCTCamera* SGCTCamera::clone() const { return new SGCTCamera(*this); }
+
+std::string SGCTCamera::getClassIdentifier() const { return classIdentifier; }
+
+const std::string SGCTCamera::classIdentifier = "SGCTCamera";
+
+void SGCTCamera::updateFrom(const Camera& source) {
+    Camera::updateFrom(source);
+
+    if (auto sc = dynamic_cast<const SGCTCamera*>(&source)) {
+        bool modified = false;
+
+        if (fovy_ != sc->fovy_) {
+            fovy_ = sc->fovy_;
+            invalidateProjectionMatrix();
+            modified = true;
+        }
+
+        if (extProj_ != sc->extProj_) {
+            extProj_ = sc->extProj_;
+            invalidateProjectionMatrix();
+            modified = true;
+        }
+
+        if (extView_ != sc->extView_ || extModel_ != sc->extModel_) {
+            extView_ = sc->extView_;
+            extModel_ = sc->extModel_;
+            invalidateViewMatrix();
+            modified = true;
+        }
+
+        if (modified && camprop_) {
+            camprop_->propertyModified();
+        }
+
+    } else if (auto pc = dynamic_cast<const PerspectiveCamera*>(&source)) {
+        setFovy(pc->getFovy());
+    } else if (auto spc = dynamic_cast<const SkewedPerspectiveCamera*>(&source)) {
+        setFovy(spc->getFovy());
+    } else if (auto oc = dynamic_cast<const OrthographicCamera*>(&source)) {
+        const auto dist = util::widthToViewDist(oc->getWidth(), getFovy(), getAspectRatio());
+        setLookFrom(getLookTo() + dist * glm::normalize(getLookFrom() - getLookTo()));
+    }
+}
+
+void SGCTCamera::setFovy(float val) {
+    if (fovy_ != val) {
+        fovy_ = val;
+        invalidateProjectionMatrix();
+        if (camprop_) {
+            if (auto p = util::getCameraFovProperty(*camprop_)) {
+                p->propertyModified();
+            }
+        }
+    }
+}
+
+void SGCTCamera::configureProperties(CameraProperty& cp, bool attach) {
+    Camera::configureProperties(cp, attach);
+    if (attach) {
+        util::updateOrCreateCameraFovProperty(
+            cp, [this]() { return getFovy(); }, [this](const float& val) { setFovy(val); });
+
+    } else if (auto fov = util::getCameraFovProperty(cp)) {
+        fov->setGetAndSet([val = fov->get()]() { return val; }, [](const float&) {});
+    }
+}
+
+bool SGCTCamera::equal(const Camera& other) const {
+    if (auto rhs = dynamic_cast<const SGCTCamera*>(&other)) {
+        return equalTo(other) && fovy_ == rhs->fovy_ && extProj_ == rhs->extProj_ &&
```

Comment:
include `aspectRatio_` set in `setExternal()`?

Reply:
the perspective camera doesn't do that